marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs

```diff
@@ -687,7 +689,8 @@ function testMappingExpression(mapping, propertyName, sourceInstance, userParame
     }
   }
   catch(e){
-    resp.errorMessage = hubUtils.getErrorMessage(e);
+    const error = mappingLibrary.parseMappingError(e);
+    resp.errorMessage = typeof error  === 'string' ? error : hubUtils.getErrorMessage(e);
```

Comment:
This is repeated above, so I think it's worth making a single function for it. Also, "parseMappingError" implies that some set of things will be returned based on the parsing - that's not the case here. The use case is - I've got an error, and I want to get a user-friendly error message for it. So how about "extractUserFriendlyErrorMessage"? And that function will have a comment noting that if that cannot be done, it'll fall back to hubUtils.getErrorMessage.